mlflow/telemetry/schemas.py

```diff
@@ -1,34 +1,128 @@
+import json
 import platform
 import sys
 import uuid
-from dataclasses import dataclass
+from abc import abstractmethod
+from dataclasses import asdict, dataclass
 from enum import Enum
 from typing import Any, Optional
 
-from mlflow.telemetry.params import BaseParams
+from mlflow.telemetry.constant import PACKAGES_TO_CHECK_IMPORT
 from mlflow.version import IS_MLFLOW_SKINNY, IS_TRACING_SDK_ONLY, VERSION
 
 
-class APIStatus(str, Enum):
+class Status(str, Enum):
     UNKNOWN = "unknown"
     SUCCESS = "success"
     FAILURE = "failure"
 
 
+class EventName(str, Enum):
```

Comment:
Can we define an abstract class and extend it for each event?


```
class Event:
    @abc
    def name() -> str:
        ...

    @abc
    def extract_params() -> ...:
        ...
```

Reply:
Updated!